R/git.R

```diff
@@ -63,18 +63,52 @@ git_path <- function(git_binary_name = NULL) {
 
 # GitHub ------------------------------------------------------------------
 
-github_info <- function(path = ".") {
+uses_github <- function(path = ".") {
   if (!uses_git(path))
-    return(github_dummy)
+    return(FALSE)
 
   r <- git2r::repository(path, discover = TRUE)
-  if (!("origin" %in% git2r::remotes(r)))
+  r_remote_urls <- remote_urls(r)
+
+  any(grepl("github", r_remote_urls))
+
+  ## or require ALSO that name of matching remote be in github_remote_names?
+  #any(grepl("github", r_remote_urls) &
+  #      names(r_remote_urls) %in% github_remote_names)
+}
+
+github_info <- function(path = ".", remote_name = NULL) {
+  if (!uses_git(path))
+    return(github_dummy)
+
+  if (!uses_github(path))
     return(github_dummy)
 
-  github_remote_parse(git2r::remote_url(r, "origin"))
+  r <- git2r::repository(path, discover = TRUE)
+  r_remote_urls <- grep("github", remote_urls(r), value = TRUE)
```

Comment:
I think the worry here is that if you have both origin and something else that points to github (e.g. upstream), this will pick the wrong one.